src/Form/RenderFileTab.tsx

```diff
@@ -20,6 +25,10 @@ export default function RenderFileTab({ step, splitSchema, setSplitSchema }: Ren
     display: 'none',
   })
 
+  const buildOptionsStep: Step = splitSchema.steps.filter(
```

Comment:
This could be replaced with a `find`. Also it's worth memoing this as we shouldn't be performing array iterations unless it's needed:
```
  const buildOptionsStep = useMemo(
    () => splitSchema.steps.find((buildOptionSchemaStep) => buildOptionSchemaStep.section === 'buildOptions'),
    [splitSchema.steps]
  )
```
Again, this will now be typed as `Steps | undefined`, which is the same as doing `splitSchema.steps.filter(...)[0]` as the filter could return an empty array. You'll need to add in conditional logic to handle the optional `buildOptionsStep`, but as with before you can just conditionally return an informative message/error instead of the entire component.